### PR TITLE
fix flaky integration tests in xample-domain 

### PR DESCRIPTION
**Pull request by lisac**
## What was the problem?

The continuous Integration (CI) tests for `xample-domain` are flaky: they appear to fail randomly for no apparent reason. 

As reported in https://github.com/department-of-veterans-affairs/abd-vro/issues/2345, this error is consistent in the logs of those failed tests:
> o.s.a.r.l.SimpleMessageListenerContainer : Failed to check/redeclare auto-delete queue(s). [...] Unexpected ERROR logs in svc-xample-j container.

Theory: 1) the `domain-xample` integration test code (java) occasionally fails to connect to the rabbitMQ service, even if the service is observed to be running*; 2) initial connection failures are not uncommon; and 3) that if given a modest number of retry attempts, the connection would succeed.

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/abd-vro/issues/2345

## How does this fix it?
1. Introduce a retry policy on rabbitMQ actions, limited to the scope of the integration test.
2. Refine error interpretation: rather than assess an integration test failure if the word "ERROR" is found in a log file, ignore cases of "ERROR" that are associated with AMQP's "SimpleMessageListenerContainer" - with the theory that initial connection failures are not uncommon. 
3. ~To mitigate the previous item being too lax, assess an integration failure if the word "Test failures:" is found in the log file.~. (test failures will in fact cause the the log file assessment to be skipped)

This PR has an additional change to the integration test, I think minor (removal of an action I thought redundant).

## How to test this PR
- in particular, please assess whether the change to the error interpretation could be more refined
- Step 2[^secrel]

[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

*the CI workflow verifies the rabbitMQ service is ready before moving on to executing the integration tests. https://github.com/department-of-veterans-affairs/abd-vro/blob/e3ec82a31b254bcb02e53b95004c844021025c67/.github/workflows/xample-integration-test.yml#L46-L50

### CODE REVIEW

**Comment by lisac** [comment]
verified that the CI fails (failure was intentional, per 93cb964) - although wasn't able to get insight into the log file handling, as that step was skipped, due to the failure detected at the integration test step. 


Will rely on code review for assessing whether the change to log file handling is acceptable.

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 1]
```diff
@@ -32,14 +34,22 @@ public class XampleJavaMicroserviceTest {
...
+    retryTemplate.setRetryPolicy(new MaxAttemptsRetryPolicy(5));
+    return retryTemplate;
   }
 
-  @AfterEach
-  private void tearDown() {
-    rabbitAdmin.purgeQueue(queueName, true);
+  @BeforeEach
+  private void setUp() {
+    rabbitTemplate.setRetryTemplate(retryTemplate);
```
Comment (lisac) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:51]: This placement of `setRetryTemplate()` is not ideal, for a few reasons: 
1. it should be executed just once, such as in a `@BeforeAll` function. However, the method signature for [`@BeforeAll`](https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/BeforeAll.html) is static, which is incompatible with the `rabbitTemplate` instance.
2. it should be configured in spring (or whatever it is that sets up `rabbitTemplate`)

... and I'm stumped on how to implement either of these. Very open to pointers. Thanks.

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 2]
```diff
@@ -32,14 +34,22 @@ public class XampleJavaMicroserviceTest {
...
+
+    RetryTemplate retryTemplate = RetryTemplate.defaultInstance();
+    retryTemplate.setBackOffPolicy(backOffPolicy);
+    retryTemplate.setRetryPolicy(new MaxAttemptsRetryPolicy(5));
+    return retryTemplate;
   }
 
-  @AfterEach
-  private void tearDown() {
-    rabbitAdmin.purgeQueue(queueName, true);
```
Comment (lisac) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:42]: i removed this method as I thought it redundant to the `purgeQueue()` performed in `setUp()`

**Review by msnwatson — APPROVED** [review]

**Comment by msnwatson** [comment]
Great questions in this PR about implementation. I know the static method stuff I've bumped into a lot and don't really have an answer and unfortunately I'm not much of a spring expert either.

**Comment by lisac** [comment]
Proceeding with merging. The implementation could be better. On the other hand, it's scoped to a test class that wouldn't affect production; and it is effective with respect to reducing flakiness. Given my limited springboot comfort, I'm not confident in the value of spending more time on this. 

Forgive me, whoever next handles this code.

**Pull request merged**